### PR TITLE
Introduce autoNamedValues macros for automatic create named values

### PR DESCRIPTION
**Pull request by greenhost87**
Introduce autoNamedValues macros for automatic create named values Map for insert query

### CODE REVIEW

File: scalikejdbc-syntax-support-macro/src/main/scala/scalikejdbc/autoNamedValues.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  def apply_impl[E: c.WeakTypeTag](c: Context)(entity: c.Expr[E], column: c.Expr[ColumnName[E]], excludes: c.Expr[String]*): c.Expr[Map[SQLSyntax, ParameterBinder]] = {
+    import c.universe._
+
+    val toMapParams: List[c.universe.Tree] = EntityUtil.constructorParams[E](c)("autoInsert", excludes: _*).map { field =>
+      val fieldName = field.name.toTermName
+
+      q"$column.$fieldName -> $entity.$fieldName"
+    }
+
+    c.Expr[Map[SQLSyntax, ParameterBinder]](q"Map(..$toMapParams)")
```
Comment (xuwei-k) [inline scalikejdbc-syntax-support-macro/src/main/scala/scalikejdbc/autoNamedValues.scala]: please use fully qualified name

```scala
c.Expr[Map[SQLSyntax, ParameterBinder]](q"_root_.scala.collection.immutable.Map(..$toMapParams)")
```

File: scalikejdbc-syntax-support-macro/src/main/scala/scalikejdbc/autoNamedValues.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package scalikejdbc
+
+import scala.language.experimental.macros
+import scala.reflect.macros.blackbox.Context
+
+object autoNamedValues {
+  def apply_impl[E: c.WeakTypeTag](c: Context)(entity: c.Expr[E], column: c.Expr[ColumnName[E]], excludes: c.Expr[String]*): c.Expr[Map[SQLSyntax, ParameterBinder]] = {
+    import c.universe._
+
+    val toMapParams: List[c.universe.Tree] = EntityUtil.constructorParams[E](c)("autoInsert", excludes: _*).map { field =>
```
Comment (seratch) [inline scalikejdbc-syntax-support-macro/src/main/scala/scalikejdbc/autoNamedValues.scala]: `"autoNamedValues"` instead of `"autoInsert"` here?

**Comment by seratch** [comment]
Looks useful. 👍

File: scalikejdbc-syntax-support-macro/src/main/scala/scalikejdbc/autoNamedValues.scala (hunk 2 above)
Comment (greenhost87) [inline scalikejdbc-syntax-support-macro/src/main/scala/scalikejdbc/autoNamedValues.scala]: yes, you right

**Pull request merged**